modules/client/src/main/java/org/apache/ignite/jdbc/ConnectionProperties.java

```diff
@@ -0,0 +1,444 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.jdbc;
+
+import java.sql.SQLException;
+import org.apache.ignite.internal.client.HostAndPortRange;
+import org.apache.ignite.internal.client.ProtocolBitmaskFeature;
+
+/**
+ * Provide access and manipulations with connection JDBC properties.
+ */
+public interface ConnectionProperties {
+    /** SSL mode: DISABLE. */
+    String SSL_MODE_DISABLE = "disable";
+
+    /** SSL mode: REQUIRE. */
+    String SSL_MODE_REQUIRE = "require";
```

Comment:
I always thought there are 3 possible values: disable, want, required.
"want" will use SSL if another part also supports it, otherwise fallback to non-SSL mode.
"required" is a mandatory requirement for SSL support to another part, otherwise connection will NOT be established.

Reply:
I created separate task IGNITE-15428 for SSL connection. I suggest make all ssl-related changes in a separate ticket.